12-offer-encoding.md

```diff
@@ -551,144 +585,100 @@ using the `onion_message` `invoice` field.
       if it has a preference.
     - for the bitcoin chain, it MUST set each `fallback_address` with
       `version` as a valid witness version and `address` as a valid witness
       program
-  - MUST include `paths` containing one or more paths to the node.
-    - MUST specify `paths` in order of most-preferred to least-preferred if it has a preference.
-    - MUST include `blindedpay` with exactly one `blinded_payinfo` for each `blinded_path` in `paths`, in order.
-    - if it includes `blinded_capacities`:
-      - MUST include exactly one `u64` (in millisatoshis) per `blinded_path`, reflecting the expected maximum amount that can be sent through the path.
+  - MUST include `invoice_paths` containing one or more paths to the node.
+    - MUST specify `invoice_paths` in order of most-preferred to least-preferred if it has a preference.
+    - MUST include `invoice_blindedpay` with exactly one `blinded_payinfo` for each `blinded_path` in `paths`, in order.
+    - MUST set `features` in each `blinded_payinfo` to match `encrypted_data_tlv`.`allowed_features` (or empty, if no `allowed_features`).
     - SHOULD ignore any payment which does not use one of the paths.
-  - MUST specify `amount`.`msat` in multiples of the minimum lightning-payable unit
-    (e.g. milli-satoshis for bitcoin) for `chain` (or for bitcoin, if there is no `chain`).
   - if responding to an `invoice_request`:
-    - if for the same `offer_id` and `payer_key` as a previous `invoice_request`:
+    - if `invoice_request_payer_key` and offer are identical to a previous `invoice_request`:
       - MAY simply reuse the previous invoice.
     - otherwise:
       - MUST NOT reuse a previous invoice.
-    - MUST set `node_id` the same as the offer.
-    - MUST set (or not set) `quantity` exactly as the invoice_request did.
-    - MUST set `payer_key` exactly as the invoice_request did.
-    - MUST set (or not set) `payer_info` exactly as the invoice_request did.
-    - MUST set (or not set) `payer_note` exactly as the invoice_request did,
-      or MUST not set it.
-    - MUST set (or not set) `replace_invoice` exactly as the invoice_request did.
-    - MUST begin `description` with the `description` from the offer.
-    - MAY append additional information to `description` (e.g. " +shipping").
-    - MUST set `amount` to the *base invoice amount* calculated from the invoice_request
-    - MUST set (or not set) `issuer` exactly as the offer did.
-    - MUST NOT set `refund_for`
-    - MUST NOT set `refund_signature`
-  - otherwise (responding to a `send_invoice` offer):
-    - MUST set `node_id` to the id of the node to send payment to.
-    - MUST set `description` the same as the offer.
-    - if the offer had a `quantity_min` or `quantity_max` field:
-      - MUST set `quantity`
+    - if `invoice_request_amount` is not present:
+      - MUST set `invoice_request_amount`.`msat` to the *base invoice amount* calculated from the invoice_request
```

Comment:
Again, I think we should not touch the invoice request part when creating the invoice. Just add a new field `invoice_amount`.